components/raftstore-v2/src/operation/command/admin/split.rs

```diff
@@ -0,0 +1,884 @@
+                }
+            }
+        }
+
+        drop(meta);
+
+        let ranges_to_delete = get_range_not_in_region(&region);
+        tablet
+            .delete_ranges_cfs(DeleteStrategy::DeleteFiles, &ranges_to_delete)
+            .unwrap_or_else(|e| {
+                error!(self.logger,"failed to delete files in range"; "err" => %e);
+            });
+
+        if last_split_region {
+            // todo: cfg validation should be called to make
+            // region_split_check_diff be Some(x)
+
+            // To prevent from big region, the right region needs run split
+            // check again after split.
+            // new_peer
+            //     .peer_mut()
+            //     .set_size_diff_hint(store_ctx.cfg.
+            // region_split_check_diff().0);
+        }
+
+        store_ctx.router.force_send(
+            source_state.get_region().id,
+            PeerMsg::AcrossPeerMsg(AcrossPeerMsg::SplitRegionInitResp(Box::new(
+                SplitRegionInitResp {
+                    parent_state: source_state,
+                    child_region_id: region_id,
+                    result: true,
+                },
+            ))),
+        );
+
+        if need_schedule_apply_fsm {
+            self.schedule_apply_fsm(store_ctx);
+        }
+    }
+
+    pub fn handle_peer_split_response<T>(
+        &mut self,
+        store_ctx: &mut StoreContext<EK, ER, T>,
+        resp: Box<SplitRegionInitResp>,
+    ) {
+        let SplitRegionInitResp {
+            parent_state,
+            child_region_id,
+            result,
+        } = Box::into_inner(resp);
+
+        assert_eq!(parent_state, *self.storage().region_state());
+
+        let mut split_progress = self.split_progress_mut();
+        *split_progress.get_mut(&child_region_id).unwrap() = true;
+
+        if split_progress.values().all(|v| *v) {
+            // Split can be finished
+            let mut wb = store_ctx.engine.log_batch(5);
+            let state = self.storage().region_state();
+            wb.put_region_state(self.region_id(), state)
+                .unwrap_or_else(|e| {
+                    panic!(
+                        "{:?} fails to update region {:?}: {:?}",
+                        self.logger.list(),
+                        state,
+                        e
+                    )
+                });
+            store_ctx.engine.consume(&mut wb, true).unwrap();
+
+            let region_id = self.region_id();
+            let tablet_index = self.storage().region_state().tablet_index;
+            let split_tablet_path = store_ctx
+                .tablet_factory
+                .split_tablet_path(region_id, tablet_index);
+            let tablet = store_ctx
+                .tablet_factory
+                .load_tablet(&split_tablet_path, region_id, tablet_index)
+                .unwrap();
+
+            let ranges_to_delete = get_range_not_in_region(self.region());
+            // todo: async version
+            tablet
```

Comment:
>  let ranges_to_delete = get_range_not_in_region(self.region());
>             // todo: async version
>             tablet
>                 .delete_ranges_cfs(DeleteStrategy::DeleteFiles, &ranges_to_delete)
>                 .unwrap_or_else(|e| {
>                     error!(self.logger,"failed to delete files in range"; "err" => %e);
>                 });

I think there's no point to delete it here as it does not delete all the unwanted data and it blocks the peer's serving requests. Instead, using compaction filter is enough.